src/luis-api-client.ts

```diff
@@ -14,252 +14,496 @@
+        body?: any;
+    }
+}
 
-const request = require('request-promise-native');
+export namespace LuisApi {
+    export interface AppInfo {
+        id: string;
+        name: string;
+        description: string;
+        culture: string;
+        active: boolean;
+        numberOfIntents: number;
+        numberOfEntities: number;
+        isTrained: boolean;
+    }
 
-export namespace Luis {
-    export interface UtteranceEntity {
-        entity: string;
-        startPos: number;
-        endPos: number;
+    export interface IntentClassifier {
+        id: string;
+        name: string;
     }
 
-    export interface Utterance {
-        text: string;
-        intent: string;
+    export interface EntityExtractor {
+        id: string;
+        name: string;
+    }
+
+    export enum PhraseListModes { Exchangeable, NonExchangeable }
+
+    export interface PhraseList {
         id?: string;
-        entities: UtteranceEntity[];
+        name: string;
+        mode: PhraseListModes;
+        isActive?: boolean;
+        editable?: boolean;
+        phrases: string;
+    }
+
+    export interface LabeledEntity {
+        name: string;
+        startToken: number;
+        endToken: number;
+        word: string;
+        isBuiltInExtractor: boolean;
     }
 
-    export interface UpdateUtteranceResult {
+    export interface LabeledUtterance {
+        id: string;
         utteranceText: string;
-        has_error: boolean;
-        error: string;
+        intent: string;
+        entities: LabeledEntity[];
     }
 
-    export interface TrainingStatusDetails {
-        statusId: number;
-        status: string;
-        trainingTime: string;
-        exampleCount: number;
-        trainingDuration: string;
-        failureReason: string;
+    export interface Entity {
+        entityType: string;
+        startToken: number;
+        endToken: number;
+    }
+
+    export interface Example {
+        exampleText: string;
+        selectedIntentName: string;
+        entityLabels?: Entity[];
     }
 
-    export interface TrainingStatus {
+    export enum TrainingStatuses { UpToDate = 2, InProgress = 3 }
+
+    export interface ModelTrainingStatus {
         modelId: string;
-        details: TrainingStatusDetails;
+        status: TrainingStatuses;
+        exampleCount: number;
+        failureReason?: string;
     }
 
-    export interface AppPublishData {
+    export type TrainingStatus = ModelTrainingStatus[];
+
+    export interface PublishResult {
         url: string;
-        previewUrl: string;
         subscriptionKey: string;
-        publishDate: string;
+        publishDate: Date;
     }
 }
 
-export class LuisClient {
-    BASE_API_PATH = 'https://api.projectoxford.ai/luis/v1.0/prog/apps/';
-    _baseRequest: any = null;
-    DEFAULT_PAGE_SIZE = 100;
+const RETRY_OPTS = {
+    retries: 10,
+    factor: 2,
+    minTimeout: 1500
+};
+// Maximum number of parallel requests to the API (to minimize 429 rate errors)
```

Comment:
Yes, it is ok. It is going to support up to 4 requests per second. I'll add "per second" to the comment to make it clearer.